### PR TITLE
[Balance] Rework Multi-Lens

### PR DESCRIPTION
**Pull request by innerthunder**
> [!IMPORTANT]
> This PR requires changes to Multi-Lens' description and therefore requires Senior Translator approval

## What are the changes the user will see?
Multi-Lens has been reworked to improve game balance:
- It now adheres to the same restrictions as [Parental Bond](https://bulbapedia.bulbagarden.net/wiki/Parental_Bond_(Ability)) when boosting attacks, meaning it no longer boosts multi-hit moves, charge moves, self-sacrificial moves, or multi-target moves (unless they only hit 1 target).
- Its maximum stack count has been reduced from 3 to 2.
- It now stacks *additively* with Parental Bond instead of *multiplicatively*, meaning that the highest achievable hit count for any non-multi-strike move is now 4 (2 Multi-Lenses + Parental Bond)
- It's power reduction of 60/75% has been changed to a damage multiplier. The first hit of an attack boosted by Multi-Lens deals $1 - 0.25m$ times the base attack damage, where $m$ is the number of held Multi-Lenses. Subsequent strikes deal 25% of the base attack damage. For example:
  - A move boosted by 1 Multi-Lens will deal 75% damage on the first strike, then 25% damage on the second strike
  - 2 Multi-Lens will yield 50% + 25% + 25% damage
  - 2 Multi-Lens with Parental Bond will yield 50% + 25% + 25% + 25%, or 125% combined damage.

## Why am I making these changes?
Requested by Balance Team. Multi-Lens' current iteration is extremely polarizing -- in the worst case, it's a net negative due to the overall power reduction, and in the best case, it enables uninteractive strategies where the enemy has virtually no chance to move. Its current form also poses many problems with respect to game design and development. Any new item idea that introduces a new chance-based on-hit effect has to be heavily scrutinized because the combination of multi-hit moves and stacked Multi-Lenses virtually guarantees the effect triggering at least once, if not multiple times.

These changes to Multi-Lens seek to improve the item's power level in the worst case while placing restrictions on some of the more overpowered use cases for the item.

## What are the changes from a developer perspective?
- `data/move`:
  - Moved Parental Bond's `canApply` conditions to a method in `Move` to more easily share conditions with Multi-Lens
  - Multi-Lens now applies its multiplier to damage in `Pokemon.getAttackDamage` instead of in `Move.calculateBattlePower`
- `modifier/modifier`: Rewrote Multi-Lens' modifier code:
  - Now enforces Parental Bond's application restrictions
  - Now boosts attacks' hit count additively
  - No longer restricts the boost on fixed-damage moves
- Cleaned up Parental Bond and Multi-Lens to use `null` in place of magic `NumberHolder(0)` args, among other small changes.
- Removed the restriction against fixed damage moves when applying Multi-Lens in `MoveEffectPhase`
- `test/abilities/parental_bond.test`: 2 tests edited and 2 tests removed reflecting changes to the interaction between Parental Bond and Multi-Lens

## How to test the changes?
`npm run test parental_bond` (some test edits)
`npm run test multi_lens` (WIP)

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [?] Are the changes visual?
  - [?] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/ability.ts [hunk 1]
```diff
@@ -1351,65 +1351,30 @@ export class AddSecondStrikeAbAttr extends PreAttackAbAttr {
...
-    const numTargets = args[0] as integer;
-    const hitCount = args[1] as Utils.IntegerHolder;
-    const multiplier = args[2] as Utils.NumberHolder;
+    const hitCount = args[0] as Utils.NumberHolder;
+    const multiplier = args[1] as Utils.NumberHolder;
 
-    if (this.canApplyPreAttack(move, numTargets)) {
+    if (move.canBeMultiStrikeEnhanced(pokemon)) {
       this.showAbility = !!hitCount?.value;
       if (!!hitCount?.value) {
```
Comment (DayKev) [inline src/data/ability.ts:1372]:
```suggestion
      if (hitCount?.value) {
```

File: src/data/ability.ts [hunk 2]
```diff
@@ -1351,65 +1351,30 @@ export class AddSecondStrikeAbAttr extends PreAttackAbAttr {
...
-        hitCount.value *= 2;
+        hitCount.value += 1;
       }
 
-      if (!!multiplier?.value && pokemon.turnData.hitsLeft % 2 === 1 && pokemon.turnData.hitsLeft !== pokemon.turnData.hitCount) {
-        multiplier.value *= this.damageMultiplier;
+      if (!!multiplier?.value && pokemon.turnData.hitsLeft === 1) {
```
Comment (DayKev) [inline src/data/ability.ts:1376]:
```suggestion
      if (multiplier?.value && pokemon.turnData.hitsLeft === 1) {
```

**Review by DayKev — COMMENTED** [review]
Other than the few tests that need to be reworked it seems good to me.

**Review by damocleas — APPROVED** [review]

**Review by frutescens — DISMISSED** [review]
reference to spaghetti

**Review by DayKev — APPROVED** [review]

**Review by damocleas — APPROVED** [review]

**Comment by Snailman11** [comment]
Paves over (Parental Bond + Multi Lens - extra hits are getting the PB modifier after one opposing mon is killed)
https://discord.com/channels/1125469663833370665/1259979182725664788

And might sidestep an existing issue with Dancer and Multilens copying
(Dancer is Buggy with Multi Lens)
https://discord.com/channels/1125469663833370665/1293467324141273088

**Review by DayKev — APPROVED** [review]

**Review by damocleas — DISMISSED** [review]

**Review by frutescens — APPROVED** [review]

**Review by damocleas — DISMISSED** [review]

**Comment by innerthunder** [comment]
Locale has been updated:

image

**Review by damocleas — APPROVED** [review]
this is the third time asking for the approval smh

**Review by DayKev — APPROVED** [review]

**Review by Adri1 — APPROVED** [review]

**Review by SangaraSorama — APPROVED** [review]
Approving as Head of Translation, the locales PR is merged. Not a dev review.

**Pull request merged**